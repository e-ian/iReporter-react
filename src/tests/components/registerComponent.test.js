import React from "react";
import { shallow } from "enzyme/build";
import RegisterComponent from "../../components/registerComponent";

describe("register component tests", () => {
  let wrapper;
  let props = {
    firstname: "",
    lastname: "",
    role: "",
    username: "",
    email: "",
    password: "",
    errors: {
      username: "",
      email: "",
      password: ""
    },
    message: {
      errors: {}
    },
    classes: null,
    handleOnChange: jest.fn(),
    handleOnSubmit: jest.fn()
  };
  beforeEach(() => {
    wrapper = shallow(<RegisterComponent {...props} />);
  });
  it("should render with crushing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should handle  onchange user input fields ", () => {
    wrapper
      .dive()
      .find("#username")
      .simulate("change");
    expect(props.handleOnChange).toHaveBeenCalledTimes(1);
  });
  it("should simulate onclick event for submit button", () => {
    const onSubmit = wrapper.dive();
    onSubmit.find(".button-success").simulate("submit");
  });
});
