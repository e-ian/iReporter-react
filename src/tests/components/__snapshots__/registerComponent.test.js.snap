// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register component tests should render with crushing 1`] = `ShallowWrapper {}`;
